Base/Python/slicer/parameterNodeWrapper/serializers.py

```diff
@@ -970,3 +966,50 @@ def supportsCaching(self):
         not update the underlying parameter node, which can be confusing.
         """
         return all([s.supportsCaching() for s in self._serializers])
+
+
+@parameterNodeSerializer
+class EnumSerializer(Serializer):
+    @staticmethod
+    def canSerialize(type_) -> bool:
+        return issubclass(type_, enum.Enum)
+
+    @staticmethod
+    def create(type_):
+        return EnumSerializer(type_)
+
+    def __init__(self, enum_):
+        self.enum = enum_
+
+        if len(self.enum) == 0:
+            raise ValueError('Cannot serialize empty Enum.')
+
+    def default(self):
+        # There is no idiomatic way to get a default value of an arbitrary enum,
+        # so just take the first item in iteration order.
+        return next(iter(self.enum))
+
+    def isIn(self, parameterNode, name: str) -> bool:
+        return parameterNode.HasParameter(name)
+
+    def write(self, parameterNode, name: str, value) -> None:
+        if not isinstance(value, self.enum):
+            raise ValueError(f'{value!r} is not a value in {self.enum!r}')
+
+        key = value.name
+        parameterNode.SetParameter(name, key)
+
+    def read(self, parameterNode, name: str):
+        key = parameterNode.GetParameter(name)
+
+        try:
+            return self.enum[key]
+        except KeyError as ex:
+            raise ValueError(f'{value!r} is not a value in {self.enum!r}') from ex
```

Comment:
Indeed. I am also updating the message to indicate the key was found in the parameter node but is not present in the enum; it might indicate the enum was changed and an old paremeter node is being read.

Reply:
Note another cause could be someone just writing random values to the name in the underlying parameter node. In general, the `read` functions are assuming that this has not happened and if it does the behavior is undefined.